test/jdk/java/awt/print/Dialog/PrintDlgPageable.java

```diff
@@ -26,104 +26,121 @@
-                return;
-            }
-            arg = Integer.parseInt(args[0]);
-                PrintDlgPageable pd = new PrintDlgPageable();
-                PrinterJob pj = PrinterJob.getPrinterJob();
-                PageableHandler handler = new PageableHandler();
-                pj.setPageable(handler);
-
-                System.out.println("open PrintDialog..");
-                if (pj.printDialog()) {
-                        try {
-                                System.out.println("About to print the data ...");
-                                pj.print();
-                                System.out.println("Printed");
-                        }
-                        catch (PrinterException pe) {
-                                pe.printStackTrace();
-                        }
-                }
+        arg = Integer.parseInt(args[0]);
+
+        String INSTRUCTIONS = " A pagedialog will be shown.";
 
+        if (arg == 0) {
+            INSTRUCTIONS += "\n Confirm that page range is disabled.";
+
+        } else if (arg == 2) {
+            INSTRUCTIONS += "\n Confirm ToPage is populated with pagerange 2";
         }
+        INSTRUCTIONS += "\nCancel the print dialog. Press PASS if it so seen else press FAIL.";
+
+        PrintDlgPageable pd = new PrintDlgPageable();
+        PrinterJob pj = PrinterJob.getPrinterJob();
+        PageableHandler handler = new PageableHandler();
+        pj.setPageable(handler);
 
-        //printable interface
-        public int print(Graphics g, PageFormat pf, int pi) throws
-PrinterException {
-
-                /*if (pi > 0) {
-                        System.out.println("pi is greater than 0");
-                        return Printable.NO_SUCH_PAGE;
-                }*/
-                // Simply draw two rectangles
-                Graphics2D g2 = (Graphics2D)g;
-                g2.setColor(Color.black);
-                g2.translate(pf.getImageableX(), pf.getImageableY());
-                g2.drawRect(1,1,200,300);
-                g2.drawRect(1,1,25,25);
-                System.out.println("print method called "+pi + " Orientation "+pf.getOrientation());
-                return Printable.PAGE_EXISTS;
+        PassFailJFrame passFailJFrame = PassFailJFrame.builder()
+            .title("Instructions")
+            .instructions(INSTRUCTIONS)
+            .rows((int) INSTRUCTIONS.lines().count() + 2)
+            .columns(35)
+            .build();
+
+        System.out.println("open PrintDialog..");
+        if (pj.printDialog()) {
+            try {
+                System.out.println("About to print the data ...");
+                pj.print();
+                System.out.println("Printed");
+            }
+            catch (PrinterException pe) {
+                pe.printStackTrace();
+            }
         }
+        passFailJFrame.awaitAndCheck();
+    }
+
+    //printable interface
+    public int print(Graphics g, PageFormat pf, int pi) throws PrinterException {
```

Comment:
`throws PrinterException` not required.

Reply:
it is required...See java.awt.print.Printable spec